packages/meditrak-server/src/tests/routes/GETDashboardGroups/assertDashboardGroupsPermissions.test.js

```diff
@@ -132,214 +130,98 @@ describe('Permissions checker for GETDashboardGroups', async () => {
+      const { body: result } = await app.get(`dashboardGroups/${projectLevelDashboardGroup.id}`);
 
-      expect(results.map(r => r.id)).to.deep.equal([nationalDashboardGroup2.id]);
+      expect(result).to.have.keys('error');
     });
   });
 
-  describe('assertDashboardGroupsPermissions()', async () => {
-    it('Sufficient permissions: Should return true if users have access to all the countries of the dashboard groups', async () => {
-      const accessPolicy = new AccessPolicy(DEFAULT_POLICY);
-      const result = await assertDashboardGroupsPermissions(accessPolicy, models, [
-        nationalDashboardGroup1,
-        nationalDashboardGroup2,
-      ]);
-
-      expect(result).to.true;
-    });
-
-    it('Sufficient permissions: Should return true if dashboard groups are project level and users have access to any of their child countries', async () => {
-      const accessPolicy = new AccessPolicy(DEFAULT_POLICY);
-      const result = await assertDashboardGroupsPermissions(accessPolicy, models, [
-        nationalDashboardGroup2,
-        projectLevelDashboardGroup,
-      ]);
-      expect(result).to.true;
-    });
-
-    it('Sufficient permissions: Should always return true for any dashboard groups if users have BES Admin access to any countries', async () => {
-      const accessPolicy = new AccessPolicy(BES_ADMIN_POLICY);
-      const results = await assertDashboardGroupsPermissions(accessPolicy, models, [
-        facilityDashboardGroup1,
-        districtDashboardGroup1,
-        nationalDashboardGroup1,
-        nationalDashboardGroup2,
-        projectLevelDashboardGroup,
-      ]);
-
-      expect(results).to.true;
-    });
-
-    it("Insufficient permissions: Should filter out any sub national dashboard groups that users do not have access to their entities' countries", async () => {
-      //Remove Admin permission of KI to have insufficient permissions to access facilityDashboardGroup1.
+  describe('GET /dashboardGroups', async () => {
+    it('Sufficient permissions: Return only the list of entries we have permissions for', async () => {
       const policy = {
         DL: ['Public'],
-        KI: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, /*'Admin'*/ 'Public'],
+        KI: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, /* 'Admin', */ 'Public'],
         SB: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Royal Australasian College of Surgeons'],
         VU: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Admin'],
-        LA: ['Admin', 'Public'],
+        LA: ['Admin'],
         TO: ['Admin'],
       };
-      const accessPolicy = new AccessPolicy(policy);
+      await prepareStubAndAuthenticate(app, policy);
+      const { body: results } = await app.get(`dashboardGroups`);
 
-      expect(() =>
-        assertDashboardGroupsPermissions(accessPolicy, models, [
-          facilityDashboardGroup1,
-          nationalDashboardGroup2,
-        ]),
-      ).to.throw;
+      expect(results.map(r => r.id)).to.deep.equal([
```

Comment:
> Test for presence and absence of dummy data, rather than full array match

I think that's just swapping one assumption about the database state for another. Since this is running as a full request, it includes pagination so we couldn't guarantee that the test data would exist in the first X (100?) entries.

I'll take a look into alternatives.

Reply:
I've updated these tests with the addition of filters to only request the test data. We discussed in person the need to add some unit tests for that filter function so I've written a separate ticket for that work: https://github.com/beyondessential/tupaia-backlog/issues/1490